PocketKit/Sources/PocketKit/Settings/AccountViewModel.swift

```diff
@@ -98,19 +98,29 @@ class AccountViewModel: ObservableObject {
         userManagementService.logout()
     }
 
-    func toggleAppBadge() {
-        UNUserNotificationCenter.current().requestAuthorization(options: .badge) {
+    func toggleAppBadge(to isEnabled: Bool) {
+        tracker.track(event: Events.Settings.appBadgeToggled(newValue: isEnabled))
+
+        UNUserNotificationCenter.current().requestAuthorization(options: .badge) { [weak self]
             (granted, error) in
+            guard let self else { return }
+
             guard error == nil && granted == true else {
+                if let error {
+                    Log.capture(error: error)
+                }
+
+                tracker.track(event: Events.Settings.appBadgePermissionDenied())
+
                 self.userDefaults.set(false, forKey: AccountViewModel.ToggleAppBadgeKey)
                 DispatchQueue.main.async { [weak self] in
                     self?.appBadgeToggle = false
                 }
```

Comment:
curious to understand why we need to set this as opposed to using the userDefaults as the source of truth to update the view. just a thought, not a blocker

Reply:
appBadgeToggle is observed by the Toggle UI element.
I can't think of how we might bind a UserDefault value to a ui component.